google-cloud-spanner/src/main/java/com/google/cloud/spanner/BatchClientImpl.java

```diff
@@ -256,15 +288,30 @@ public List<Partition> partitionQuery(
       builder.setPartitionOptions(pbuilder.build());
 
       final PartitionQueryRequest request = builder.build();
-      PartitionResponse response = rpc.partitionQuery(request, options);
-      ImmutableList.Builder<Partition> partitions = ImmutableList.builder();
-      for (com.google.spanner.v1.Partition p : response.getPartitionsList()) {
-        Partition partition =
-            Partition.createQueryPartition(
-                p.getPartitionToken(), partitionOptions, statement, queryOptions);
-        partitions.add(partition);
+      try {
+
+        PartitionResponse response = rpc.partitionQuery(request, options);
+        ImmutableList.Builder<Partition> partitions = ImmutableList.builder();
+        for (com.google.spanner.v1.Partition p : response.getPartitionsList()) {
+          Partition partition =
+              Partition.createQueryPartition(
+                  p.getPartitionToken(), partitionOptions, statement, queryOptions);
+          partitions.add(partition);
+        }
+        return partitions.build();
+      } catch (SpannerException e) {
+        maybeMarkUnimplementedForPartitionedOps(e);
+        throw e;
+      }
+    }
+
+    void maybeMarkUnimplementedForPartitionedOps(SpannerException spannerException) {
+      if (spannerException.getErrorCode() == ErrorCode.INVALID_ARGUMENT
```

Comment:
Are you sure that the error code will be `INVALID_ARGUMENT` (and not `UNIMPLEMENTED`)?

Reply:
Yes, for Partitioned Read or Query, we are returning INVALID_ARGUMENT.

Reply:
@pratickchokhani 
As discussed in the meeting, can we temporarily remove the error code check?

Context:
We plan to update the backend to return the UNIMPLEMENTED error for partitioned operations. However, this change will not be part of the current release and will be introduced in a follow-up update.

If we proceed with this error code check now, it will break backward compatibility when the backend starts returning UNIMPLEMENTED. To avoid this issue, we are removing the check for now and will reintroduce it once the backend update is complete.

cc: @rahul2393 to do the same for golang.